### ISSUE TITLE
TreeDropdownField()->setTreeBaseID broken for assets

### ISSUE THREAD

**Issue by Juanitou** [silverstripe/silverstripe-cms#2205]
Maybe related to #2112.

To reproduce:
- `composer create-project silverstripe/installer issue-2112 4.1.1`
- Run the installer.
- In the CMS Files section, create a Folder/SubFolder/SubSubFolder structure.
- Add in mysite/code/Page.php:
```php
use SilverStripe\CMS\Model\SiteTree;
use SilverStripe\Assets\Folder;
use SilverStripe\Forms\TreeDropdownField;

class Page extends SiteTree
{
    private static $db = [
        'FolderID' => 'Int'
    ];

    private static $has_one = [
        'AnotherFolder' => Folder::class
    ];

    public function getCMSFields()
    {
        $fields = parent::getCMSFields();

        // get Folder object
        $folder = Folder::get()->filter('Name', 'Folder')->First();
        if (! $folder) {
             return $fields;
        }

        // add dropdown field with folder (as $db) hierarchy
        $folderTree = new TreeDropdownField(
            'FolderID',
            'Choose folder based on ID ($db)',
            Folder::class
        );

        // add dropdown field with folder hierarchy
        $folderTree->setTreeBaseID((int) $folder->ID);
        $fields->addFieldToTab('Root.Main', $folderTree, 'Content');

        // get another Folder object
        $anotherFolder = Folder::get()->filter('Name', 'Folder')->First();
        if (! $anotherFolder) {
             return $fields;
        }

        // add dropdown field with another folder (as $has_one) hierarchy
        $anotherTree = new TreeDropdownField(
            'AnotherFolderID',
            'Choose folder based on Folder::class ($has_one)',
            Folder::class
        );

        $anotherTree->setTreeBaseID((int) $anotherFolder->ID);
        $fields->addFieldToTab('Root.Main', $anotherTree, 'Content');

        return $fields;
    }
}
```
- Try to select SubSubFolder in the db-based or has_one-based tree dropdowns, or otherwise navigate the folder hierarchy.
- So, setting the BaseID seems to work for the first field, as the proposed choice of folder is 'SubFolder'. The second field does not propose a choice. In both cases we cannot navigate the folder hierarchy.
- As soon as you comment out the setTreeBaseID() lines, both dropdowns recover their functionality.
- I have also tried with File::class, with the same results.

I hope I’m not doing something silly and it helps!

**Comment by Juanitou** [comment]
Hi! I’m still getting this in SilverStripe 4.3.3.

**Comment by mxmclaudel** [comment]
Hi,
I'm having the same issue with TreeMultiselectField for a TaxonomyTerm selection.
The base item is a root item (ParentID = 0), the list does show all its items, however if I want to display the items under one of these elements, nothing is displayed. Searching doesn't work either.

Using SilverStripe 4.4.3

**Comment by gdelavil** [comment]
Hi! I'm still getting this issue on SilverStripe 4.5.1, except it's on SiteTree objects, not classes.

Using `setTreeBaseID` on a `TreeDropdownField` with anything other than 0 will cause it to only render the initial list of pages, but then if you navigate down through the tree you can't come back up. Expanding the dropdown afterwards yields no results, and a refresh is needed before anything is available.

@tractorcow do you know if there's plans to fix this anytime soon? I would've loved to have been able to use `setTreeBaseID`.

**Comment by maxime-rainville** [comment]
I think this is the same issue as https://github.com/silverstripe/silverstripe-admin/issues/954 which just got fixed and will ship in the upcoming 4.8.

Let me know if it's different and I'll close it.
